easybuild/easyconfigs/d/DL_POLY_Classic/DL_POLY_Classic-1.9-foss-2015b-PLUMED-2.2.0.eb

```diff
@@ -0,0 +1,38 @@
+easyblock = 'MakeCp'
+
+name = 'DL_POLY_Classic'
+version = '1.9'
+
+homepage = 'http://ccpforge.cse.rl.ac.uk/gf/project/dl_poly_classic/'
+description = """DL_POLY Classic is a freely available molecular dynamics program developed
+ from the DL_POLY_2 package. This version does not install the java gui."""
+
+toolchain = {'name': 'foss', 'version': '2015b'}
+
+sources = ['dl_class_%(version)s.tar.gz']
+source_urls = ['http://ccpforge.cse.rl.ac.uk/gf/download/frsrelease/255/2627/']
+
+plumedversion = '2.2.0'
+versionsuffix = '-PLUMED-%s' % plumedversion
+
+dependencies = [('PLUMED', plumedversion)]
+
+# this patch needs to be regenerated for another version of PLUMED
+patches = [('PLUMED-%s.patch' % plumedversion, 1)]
```

Comment:
- Build and install `PLUMED`.
- Download the patch from the google groups, and drop it in `$EBROOTPLUMED/lib/plumed/patches/`. It might be that this can be avoid by commandline args. Didn't try it.
- Extract source of DL_POLY 1.9
- Copy the directory `source` to `srcmod`.
- Copy the file `build/MakePAR` to `srcmod/Makefile`
- While in the DL_POLY source dir, run: `plumed-patch -p --runtime dlpoly-2.20`.
- Now adjust the makefile for EB: 
```patch
diff -urN dl_class_1.9.orig/Plumed.inc dl_class_1.9/Plumed.inc
--- dl_class_1.9.orig/Plumed.inc        1970-01-01 01:00:00.000000000 +0100
+++ dl_class_1.9/Plumed.inc     2015-11-16 17:22:29.921325000 +0100
@@ -0,0 +1,3 @@
+# PLUMED: runtime installation
+PLUMED_LOAD= -L$(EBROOTPLUMED)/lib -lplumed -ldl
+PLUMED_DEPENDENCIES=
```
```patch
 BINROOT = ../execute
-CC  = gcc
+CC  := $(MPICC)
 EX = DLPOLY.X
 EXE = $(BINROOT)/$(EX)
-FC=undefined
+FC := $(MPIF90)
 SHELL=/bin/sh
 TYPE=par

@@ -54,9 +54,9 @@
 
 #================== GNU Fortran, MPI version ==============================
 gfortran:
-       $(MAKE) FC="mpif90" LD="mpif90 -o" \
-       LDFLAGS="-O2 -ffast-math" \
-       FFLAGS="-c -O2 -ffast-math" \
+       $(MAKE) FC="$(MPIF90)" LD="$(MPIF90) -o" \
+       LDFLAGS="$(LDFLAGS)" \
+       FFLAGS="-c $(FFLAGS)" \
        EX=$(EX) BINROOT=$(BINROOT) $(TYPE)
```

- And now, you can execute `make` in the `srcmod` directory.
- I also had to fix 2 compile errors because of the patching. Don't remember the files but you will see the error pop up and it's trivial to fix (moving a couple of lines)

Remember, this is only the case of `DL_POLY` with `PLUMED` (currently a versionsuffix). Without plumed its much easier. The currently block is basically the `MakeCp` block.